### DISCUSSION TITLE
Custom params for router.resource() in Adonisjs V6

### DISCUSSION THREAD

**Discussion by Heng229** [adonisjs/core#4819] [Help] [answered]
router.resource('/streaming-video', StreamingVideoController).params({
      show: 'slug'
})

The params() ain't working for changing the default param by router.resource()

The show() is still using :id as default params instead of :slug.

Please correct me if I misunderstood the implementation.

The prompt I got
{ id: 'privacy-policy' }  <====== console.log(params)
[04:52:09.720] ERROR (18660): show
    err: {
      "type": "ReferenceError",
      "message": "slug is not defined",
      "stack":
          ReferenceError: slug is not defined
              at StreamingVideoController.show (....controller.ts:57:70)
              at ContainerResolver.call (..\node_modules\@adonisjs\fold\src\resolver.ts:491:18)
    }

**Comment by RomainLanz** [comment, marked as answer]
Hey @Heng229! 👋🏻 

When defining custom names, the key in the object should match the resource you're renaming. In your case, that would be `privacy-policy` instead of `show`.

📚 https://docs.adonisjs.com/guides/basics/controllers#renaming-resource-params
